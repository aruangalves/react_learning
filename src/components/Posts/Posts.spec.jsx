import { render, screen } from '@testing-library/react';
import { Posts } from '.';

const props = {
    posts: [
        {
            id: 1,
            title: 'Post #1',
            body: 'Some text for post #1',
            cover: 'img/img1.jpg',
        },
        {
            id: 2,
            title: 'Post #2',
            body: 'Some text for post #2',
            cover: 'img/img2.jpg',
        },
        {
            id: 3,
            title: 'Post #3',
            body: 'Some text for post #3',
            cover: 'img/img3.jpg',
        },
    ],
};

describe('<Posts />', () => {
    it('should render all posts', () => {
        render(<Posts {...props} />);

        expect(screen.getAllByRole('heading', { name: /Post #/i })).toHaveLength(3);
        expect(screen.getAllByRole('img', { name: /Post #/i })).toHaveLength(3);
        expect(screen.getAllByText(/Some text for post #/i)).toHaveLength(3);
        expect(screen.getByRole('img', { name: /Post #3/i })).toHaveAttribute('src', 'img/img3.jpg');
    });

    it('should render an empty post list', () => {
        render(<Posts />);
        expect(screen.queryByRole('heading', { name: /Post #/i })).not.toBeInTheDocument();
    });

    it('should match snapshot', () => {
        const { container } = render(<Posts {...props} />);
        const { firstChild } = container;

        expect(firstChild).toMatchSnapshot();
    });
});
