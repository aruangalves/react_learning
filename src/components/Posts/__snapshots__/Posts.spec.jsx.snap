// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="Post #1"
      src="img/img1.jpg"
    />
    <div
      class="post-content"
    >
      <h2>
        Post #1
      </h2>
      <p>
        Some text for post #1
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Post #2"
      src="img/img2.jpg"
    />
    <div
      class="post-content"
    >
      <h2>
        Post #2
      </h2>
      <p>
        Some text for post #2
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Post #3"
      src="img/img3.jpg"
    />
    <div
      class="post-content"
    >
      <h2>
        Post #3
      </h2>
      <p>
        Some text for post #3
      </p>
    </div>
  </div>
</div>
`;
